compiler/rustc_monomorphize/src/partitioning/autodiff.rs

```diff
@@ -80,6 +101,20 @@ fn adjust_activity_to_abi<'tcx>(tcx: TyCtxt<'tcx>, fn_ty: Ty<'tcx>, da: &mut Vec
     }
 }
 
+fn count_scalar_fields<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>) -> usize {
+    match ty.kind() {
+        ty::Float(_) | ty::Int(_) | ty::Uint(_) => 1,
```

Comment:
What would happen to let's say `x: (&f32, f32)`? It's still below 2* pointer size. 

In general, I'm not sure about (small) Arrays, slice, RawPtr, Ref, bool, char.